pkg/querier/block_store_metrics.go

```diff
@@ -0,0 +1,220 @@
+package querier
+
+import (
+	"sync"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// This struct aggregates metrics exported by Thanos Bucket Store
+// and re-exports those aggregates as Cortex metrics.
+type tsdbBucketStoreMetrics struct {
+	// Maps userID -> registry
+	regsMu sync.Mutex
+	regs   map[string]*prometheus.Registry
+
+	// exported metrics, gathered from Thanos BucketStore
+	blockLoads            *prometheus.Desc
+	blockLoadFailures     *prometheus.Desc
+	blockDrops            *prometheus.Desc
+	blockDropFailures     *prometheus.Desc
+	blocksLoaded          *prometheus.Desc
+	seriesDataTouched     *prometheus.Desc
+	seriesDataFetched     *prometheus.Desc
+	seriesDataSizeTouched *prometheus.Desc
+	seriesDataSizeFetched *prometheus.Desc
+	seriesBlocksQueried   *prometheus.Desc
+	seriesGetAllDuration  *prometheus.Desc
+	seriesMergeDuration   *prometheus.Desc
+	resultSeriesCount     *prometheus.Desc
+	chunkSizeBytes        *prometheus.Desc
+
+	// Metrics gathered from Thanos storecache.InMemoryIndexCache
+	cacheItemsEvicted          *prometheus.Desc
+	cacheItemsAdded            *prometheus.Desc
+	cacheRequests              *prometheus.Desc
+	cacheItemsOverflow         *prometheus.Desc
+	cacheHits                  *prometheus.Desc
+	cacheItemsCurrentCount     *prometheus.Desc
+	cacheItemsCurrentSize      *prometheus.Desc
+	cacheItemsTotalCurrentSize *prometheus.Desc
+
+	// Ignored:
+	// thanos_store_index_cache_max_size_bytes
+	// thanos_store_index_cache_max_item_size_bytes
+}
+
+func newTSDBBucketStoreMetrics() *tsdbBucketStoreMetrics {
+	return &tsdbBucketStoreMetrics{
+		regs: map[string]*prometheus.Registry{},
+
+		blockLoads: prometheus.NewDesc(
+			"cortex_bucket_store_block_loads_total",
```

Comment:
Given the bucket store is an internal component for Cortex, I would be more keep to prefix the metrics with `cortex_querier_bucket_store` instead of `cortex_bucket_store`. What's your take?

Reply:
Fine with me. I'll change it.